httpserver/service_node.cpp

```diff
@@ -316,6 +345,58 @@ void ServiceNode::attach_signature(
     }
 }
 
+void ServiceNode::perform_peer_test() {
+
+    // 1. Select the tester/testee pair
+
+    std::vector<sn_record_t> members_pk = swarm_->other_nodes();
+
+    members_pk.push_back(our_address_);
+
+    std::sort(members_pk.begin(), members_pk.end());
+
+    std::cout << "peers: " << std::endl;
+    for (auto& pk : members_pk) {
+        std::cout << pk.address << ":" << pk.port << std::endl;
+    }
+    if (members_pk.size() < 2) {
+        BOOST_LOG_TRIVIAL(error)
+            << "Could not initiate peer test: swarm too small";
+        return;
+    }
+
+    uint64_t seed;
+    if (block_hash_.size() < sizeof(seed)) {
+        BOOST_LOG_TRIVIAL(error)
+            << "Could not initiate peer test: invalid block hash";
+        return;
+    }
+
+    std::memcpy(&seed, block_hash_.data(), sizeof(seed));
+    std::mt19937_64 mt(seed);
+    auto tester_idx =
```

Comment:
There are a few places that might be `const` but I'll wait for the rest of the implementation to see how they are used ;)